### ISSUE TITLE
6.10.3 ❯ 6.11.1 : Cannot use import statement outside a module.  inside jest environment

### ISSUE THREAD

**Issue by gre** [polkadot-js/api#4292]
while 6.10.3 was totally fine, we are now, again, reaching issues with 6.11.1.

```
    /Users/runner/work/ledger-live-common/ledger-live-common/node_modules/@polkadot/x-bigint/index.js:3
    import { xglobal } from '@polkadot/x-global';
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module
```

**Comment by jacogr** [comment]
There is nothing wrong (from what I can see) with either the x-blobal or x-bigint packages -

1. Both have export maps that split correctly for esm/cjs
2. cjs has only require, esm only import

e.g. x-global -

```
"exports": {
    ".": {
      "require": "./index.cjs",
      "default": "./index.js"
    },
 ```
 
 TL;DR This is not packaging related. (All the above is applied automatically for all packages build, if it breaks for one, it breaks for all)

**Comment by gre** [comment]
Thanks @jacogr , indeed looks like module related, the fact there were changes in deps probably make this issue appear in our side because we survive with [moduleNameMapper config in our jest config](https://github.com/LedgerHQ/ledger-live-common/blob/master/jest.config.ts#L27)

in case someone fall into this, we had to add this on top of our jest config to make it run in Jest:

```
    "^@polkadot/x-bigint/(.*)": "@polkadot/x-bigint/$1.cjs",
    "^@polkadot/x-bigint$": "@polkadot/x-bigint/index.cjs",
    "^@polkadot/x-noble-secp256k1$": "@polkadot/x-noble-secp256k1/index.cjs",
    "^@polkadot/x-noble-hashes/(.*)": "@polkadot/x-noble-hashes/$1.cjs",
    "^@polkadot/x-noble-hashes$": "@polkadot/x-noble-hashes/index.cjs",
    ```

(ugly hacks on our side until we find better, we didn't figured out a better way to set up Jest with the libraries so far)

**Comment by jacogr** [comment]
Jest is actually improving each bump release (each time there is a minor, I drop overrides and do try) - but yes, the ESM support is still a WIP and it is _still_ problematic.
